### ISSUE TITLE
 [SQL] Integer arithmetic causes value to wrap around instead of returning an error

### ISSUE THREAD

**Issue by abhizer** [feldera/feldera#1186]
Instead of returning an out of range error when performing arithmetic operations on integers, the values get wrapped around. 

Example: 
https://github.com/feldera/feldera/blob/a3e1b361bd817d2767d2c0a35fc2111234aaa73d/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt2Tests.java#L326

**Comment by mihaibudiu** [comment]
@abhizer for this we need to change the sqllib Rust library to use `checked` operations instead of just regular arithmetic. It looks like Rust already has the right support. So I will assign this issue to you.

**Comment by mihaibudiu** [comment]
The code is in `sqllib/src/operators.rs`

**Comment by abhizer** [comment]
#1202 proposes a trait, `SQLArithmetic`, that provides methods: 
- `plus`
- `minus`
- `mul`
- `div`

Other similar methods can be added later. 

This allows us to keep the Rust API simple, and generic over `SQLArithmetic` and the java part of the compiler doesn't need to know subtle difference in the implementation of, say addition of integers vs of floats. 

Integer types implementing `SQLArithmetic` will performed `checked` version of the operation, and floats can just use the default `+` operator. 

tldr: `SQLArithmetic` will be an interface for arithmetic operations as they are supposed to happen in SQL.

**Comment by mihaibudiu** [comment]
I left some comments about this in the PR where you give a first implementation.
The main question is about Decimal, which does not have enough information to perform checked arithmetic.
An alternative is to implement a wrapper around Decimal which carries dynamic type information (scale and precision).
